osu.Android/Properties/AndroidManifestOverlay.xml

```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
     <queries>
+        <package android:name="com.github.android" />
```

Comment:
I think this is the wrong way to fix this. This will lead to an endless game of cat-and-mouse: X links don't work so we have to add Y app to the manifest here. Apps on Android should be able to open web links without these manifest hacks. This needs fixing framework-side, probably best to use what MAUI provides.

As an example: twitter and twitch links equally don't work.

Reply:
I guess we have to allow everything. The game already does the warnings through a dialog, so we can just do this:
```diff
diff --git a/osu.Android/Properties/AndroidManifestOverlay.xml b/osu.Android/Properties/AndroidManifestOverlay.xml
index 3dec84731a..5c62b53c46 100644
--- a/osu.Android/Properties/AndroidManifestOverlay.xml
+++ b/osu.Android/Properties/AndroidManifestOverlay.xml
@@ -1,16 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
+    <permission android:name="android.permission.QUERY_ALL_PACKAGES"/>
     <queries>
-        <package android:name="com.github.android" />
         <intent>
-            <action android:name="android.intent.action.VIEW" />
-            <category android:name="android.intent.category.BROWSABLE" />
-            <data android:scheme="https" />
-        </intent>
-        <intent>
-            <action android:name="android.intent.action.VIEW" />
-            <category android:name="android.intent.category.BROWSABLE" />
-            <data android:scheme="mailto" />
+            <action android:name="android.intent.action.MAIN" />
         </intent>
     </queries>
 </manifest>
```
unless I'm missing another way that's better.

Reply:
Having the `QUERY_ALL_PACKAGES` permission probably won't work for the play store release.

The below framework-side diff seems to work for me. If we never call `intent.ResolveActivity`, there is no need for the `query` permission and manifest entries.

```diff
diff --git a/osu.Framework.Android/AndroidGameHost.cs b/osu.Framework.Android/AndroidGameHost.cs
index c6ea1c0d7..426c6214c 100644
--- a/osu.Framework.Android/AndroidGameHost.cs
+++ b/osu.Framework.Android/AndroidGameHost.cs
@@ -18,6 +18,7 @@
 using osu.Framework.Input.Handlers;
 using osu.Framework.Input.Handlers.Midi;
 using osu.Framework.IO.Stores;
+using osu.Framework.Logging;
 using osu.Framework.Platform;
 using Uri = Android.Net.Uri;

@@ -83,8 +84,14 @@ public override void OpenUrlExternally(string url)

             using (var intent = new Intent(Intent.ActionView, Uri.Parse(url)))
             {
-                if (intent.ResolveActivity(gameView.Activity.PackageManager) != null)
+                try
+                {
                     gameView.Activity.StartActivity(intent);
+                }
+                catch (ActivityNotFoundException e)
+                {
+                    Logger.Error(e, $"Failed to start intent {intent}");
+                }
             }
         }
```

Reply:
> On the other side I'm not sure about the `Logger.Error`. It'll bubble as an error to the user, not sure that's what we want.

As noted in the docs, this catch probably won't be hit.
I would rather show an error to the user, instead of having links do nothing. Makes it obvious that something isn't working and we get a report in Sentry (I would hope).